correios/client.py

```diff
@@ -99,6 +108,33 @@ def __init__(
 
         self.model_builder = ModelBuilder()
 
+    def _parse_exception(self, exception):
+        message = str(exception)
+
+        ERROR_EXCEPTIONS = (
```

Comment:
maybe using `namedtuples` will make this method easier

```python
from collections import namedtuple

CorreiosAPIError = namedtuple('CorreioAPIError', ['regex', 'exception', 'message'])

ERRORS = (
    CorreiosAPIError(exception=AuthenticationError, message='Auth error for...', regex=re.compile(r'autenticacao')),
    ...
)

for regex, exception, error_message in ERRORS:  # namedtuple maintain field ordering as specified on its creation
    if regex.search(message):
        return exception(error_message)
```

Reply:
IMO these errors contstants should be moved to the data module

Reply:
Nice suggestion @lamenezes, thanks. About the constant, if a remove from the method I lose the use of variables, like auth error message I use `self.username`.

Reply:
Concordo com o @lamenezes mas ainda não estou curtindo muito a parte da montagem da string da mensagem de erro 🙁

Mas também não consigo imaginar um jeito melhor de fazer isso. Talvez tirar todas as mensagens seja melhor do que implementar uma lógica específica para cada uma delas.

Outra opção pode ser só colocar uma cópia da mensagem de erro dos correios.

```python
raise exception(message)
```

além disso... não use `return` para exception. Use `raise`